src/launchpad/size/analysis.py

```diff
@@ -0,0 +1,28 @@
+import json
+import time
+from typing import Any, BinaryIO, TextIO, cast
+
+from ..analyzers.android import AndroidAnalyzer
+from ..analyzers.apple import AppleAppAnalyzer
+from ..artifacts import AndroidArtifact, AppleArtifact, ArtifactFactory
+
+
+def do_size_analysis(input_file: BinaryIO, output_file: TextIO) -> None:
+    start_time = time.time()
+    artifact = ArtifactFactory.from_file(input_file)
+
+    # isinstance switch below is a bit sad. Ryan suggested a
+    # get_analyzer method on artifact which might be nicer.
```

Comment:
We can't do an `Artifact.get_analyzer` unfortunately due to dependency cycles (i.e. each artifact impl depends on `Artifact`, so `Artifact` cannot import implementations. I think for now this is our best bet, but we can rethink this longer term

Reply:
Thanks! Removed this comment.